### DISCUSSION TITLE
PHP 8 attributes in v3?

### DISCUSSION THREAD

**Discussion by royvanv** [livewire/livewire#5265] [4. General]
I just read [the announcement for v3 on Laravel News](https://laravel-news.com/livewire-v3-features) and that some features use _heredoc_ notation. Is there a reason why attributes are not used here? If it's for compatibility reasons, I reckon supporting both doc notation and attributes would be nice.

So with attributes the example from Laravel would look something like this:
```php
<?php

namespace App\Http\Livewire;

use Livewire\Attributes\Prop;
use Livewire\Attributes\Js;
 
class Todos extends \Livewire\Component
{
    #[\Livewire\Attributes\Prop]
    public $todos;
 
     #[\Livewire\Attributes\Js]
    public function clear()
    {
        return <<<'JS'
            this.todo = '';
        JS;
    }
}
```

**Comment by Wulfheart** [comment]
In the talk he stated that this still has to be decided.
